DatadogRUM/Sources/RUMMonitor/Scopes/Utils/LaunchReasonResolver.swift

```diff
@@ -0,0 +1,120 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// A helper that resolves the app launch reason (`LaunchReason`) for tvOS apps, using a timed window and
+/// observed lifecycle events. This is needed because tvOS does not expose kernel-level signals (like `task_role`)
+/// that are available on iOS to detect whether an app was launched by the user or in the background.
+///
+/// It buffers RUM commands along with their corresponding `DatadogContext` and `Writer`. Once the launch reason is resolved
+/// (either by receiving a `willEnterForeground` event or exceeding the launch window threshold),
+/// it replays all buffered items in FIFO order. The original commands and writers are preserved, but the
+/// `launchReason` field in each context is updated with the resolved value before forwarding.
+internal final class LaunchReasonResolver {
+    private let window: AppLaunchWindow
+
+    init(launchWindowThreshold: TimeInterval) {
+        self.window = AppLaunchWindow(launchWindowThreshold: launchWindowThreshold)
+    }
+
+    /// Buffers the given command and forwards it once the launch reason is resolved.
+    /// If resolution occurs as a result of this command, all previously buffered commands are forwarded in order,
+    /// with their `DatadogContext.launchInfo.launchReason` updated to match the resolved reason.
+    /// If the context already contains a known `launchReason`, the command is forwarded immediately without modification.
+    func forwardWithLaunchReason(
+        command: RUMCommand,
+        context: DatadogContext,
+        writer: Writer,
+        forwardingBlock: ([(command: RUMCommand, context: DatadogContext, writer: Writer)]) -> Void
```

Comment:
suggestion/ It would be more straightforward if the `forwardingBlock` forward a single cmd. A for-each is only required when the buffer is flushed.